packages/api-gateway/test/auth.e2e-spec.ts

```diff
@@ -0,0 +1,158 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import {
+  ClassSerializerInterceptor,
+  INestApplication,
+  ValidationPipe,
+} from '@nestjs/common';
+import { AppModule } from './../src/app.module';
+import { Reflector } from '@nestjs/core';
+import * as request from 'supertest';
+import { ResetProtoFile } from 'juno-proto';
+import * as GRPC from '@grpc/grpc-js';
+import * as ProtoLoader from '@grpc/proto-loader';
+
+let app: INestApplication;
+const ADMIN_EMAIL = 'test-superadmin@test.com';
+const ADMIN_PASSWORD = 'test-password';
+
+beforeAll(async () => {
+  const proto = ProtoLoader.loadSync([ResetProtoFile]) as any;
+
+  const protoGRPC = GRPC.loadPackageDefinition(proto) as any;
+  const resetClient = new protoGRPC.juno.reset_db.DatabaseReset(
+    process.env.DB_SERVICE_ADDR,
+    GRPC.credentials.createInsecure(),
+  );
+  await new Promise((resolve) => {
+    resetClient.resetDb({}, () => {
+      resolve(0);
+    });
+  });
+});
+
+afterAll((done) => {
+  app.close();
+  done();
+});
+
+beforeEach(async () => {
+  const moduleFixture: TestingModule = await Test.createTestingModule({
+    imports: [AppModule],
+  }).compile();
+
+  app = moduleFixture.createNestApplication();
+  app.useGlobalPipes(
+    new ValidationPipe({
+      transform: true,
+    }),
+  );
+  app.useGlobalInterceptors(new ClassSerializerInterceptor(app.get(Reflector)));
+
+  await app.init();
+});
+
+describe('Invalid key push to get jwt route', () => {
+  it('Invalid email parameter to /auth/key', () => {
+    return request(app.getHttpServer())
+      .post('/auth/key')
+      .send({
+        email: 'invalid-email@gmail.com',
+        password: ADMIN_PASSWORD,
+        environment: 'prod',
+        project: {
+          name: 'test-seed-project',
+        },
+      })
+      .expect(500);
+  });
+
+  it('Invalid password parameter to /auth/key', () => {
+    return request(app.getHttpServer())
+      .post('/auth/key')
+      .send({
+        email: ADMIN_EMAIL,
+        password: 'invalid-passowrd',
+        environment: 'prod',
+        project: {
+          name: 'test-seed-project',
+        },
+      })
+      .expect(500);
+  });
+
+  it('Different environment parameter to /auth/key', () => {
```

Comment:
What is the expected difference (if any) in behavior across environments?

Reply:
If it's a valid project name, enviornment shouldn't have an issue for key gen, environments just let us define an internal split for the project (so an api key's JWT using email services with a `prod` environment won't touch the same data as an api key's JWT using email services w/ a `staging` environment.)